### PR TITLE
Fix type annotations in installer

### PR DESCRIPTION
**Pull request by kmaehashi**
Adds mypy configuration at `installer/mypy.ini`.

### CODE REVIEW

File: install/cupy_builder/install_build.py [hunk 1]
```diff
@@ -228,7 +231,7 @@ def _match_output_lines(output_lines, regexs):
     return None
 
 
-def get_compiler_base_options(compiler_path):
+def get_compiler_base_options(compiler_path) -> List[str]:
```
Comment (emcastillo) [inline install/cupy_builder/install_build.py:234]:
```suggestion
def get_compiler_base_options(compiler_path: str) -> List[str]:
```

File: install/cupy_builder/install_build.py [hunk 2]
```diff
@@ -318,7 +321,7 @@ def check_cuda_version(compiler, settings):
     return True
 
 
-def get_cuda_version(formatted=False):
+def get_cuda_version(formatted=False) -> int:
```
Comment (emcastillo) [inline install/cupy_builder/install_build.py:324]:
```suggestion
def get_cuda_version(formatted: bool = False) -> int:
```

File: install/cupy_builder/install_build.py [hunk 3]
```diff
@@ -356,7 +359,7 @@ def check_hip_version(compiler, settings):
     return True
 
 
-def get_hip_version(formatted=False):
+def get_hip_version(formatted=False) -> int:
```
Comment (emcastillo) [inline install/cupy_builder/install_build.py:362]:
```suggestion
def get_hip_version(formatted: bool = False) -> int:
```

File: install/cupy_builder/install_build.py [hunk 4]
```diff
@@ -400,7 +403,7 @@ def check_compute_capabilities(compiler, settings):
     return True
 
 
-def get_compute_capabilities(formatted=False):
+def get_compute_capabilities(formatted=False) -> Set[int]:
```
Comment (emcastillo) [inline install/cupy_builder/install_build.py:406]:
```suggestion
def get_compute_capabilities(formatted: bool = False) -> Set[int]:
```

**Review by emcastillo — COMMENTED** [review]
Some types were missing in function args 😁

File: install/cupy_builder/install_build.py (hunk 1 above)
Comment (kmaehashi) [inline install/cupy_builder/install_build.py:234]: Thanks, but surprisingly this is a list 😇 

```suggestion
def get_compiler_base_options(compiler_path: List[str]) -> List[str]:
```

(That's why we need a refactoring in installer...)

**Review by emcastillo — APPROVED** [review]

**Pull request merged**